### ISSUE TITLE
Question regarding functionality of NiftiMasker.fit_transform [Discussion]

### ISSUE THREAD

**Issue by dlevitas** [nilearn/nilearn#2717]
I'm using nilearn version `0.7.0` and have created a series of binary ROI masks that I'm using to extract the timeseries, following nuisance regression. The confound regressors have been extracted from the confound.tsv files generated by fmriprep (version `20.2.0`).  The gist of my code is:

```
roi_masker = NiftiMasker(mask_img=roi_mask_img, standardize='zscore', detrend=True, verbose=1)
nuisance_regressed_roi_timeseries = roi_masker.fit_transform(func_img, confounds=confound_df.values)
```

My assumptions are that the confounds in `fit_transform()` are first passed to `img.clean_img` before timeseries extraction, and that the signal cleaning is done in a non-modular fashion, as recommended by [Linquist et al](https://onlinelibrary.wiley.com/doi/pdfdirect/10.1002/hbm.24528). I'm curious if these two assumptions are correct, or if I need to adjust something in my code?

Thanks!

**Comment by bthirion** [comment]
Please have this type of discussion on Neurostars. The isse tracker is for ... issues.
Yes, we do this in a non-modular fashion. Normally your final data should be orthogonal to the confounds. Please let us know if you hit a problem.
Best,
